src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -285,6 +285,10 @@ object SymDenotations {
     final def addAnnotation(annot: Annotation): Unit =
       annotations = annot :: myAnnotations
 
+    /** Add given annotation to the annotations of this denotation */
+    final def addAnnotation(annot: ClassSymbol)(implicit ctx: Context): Unit =
+      addAnnotation(ConcreteAnnotation(tpd.ref(annot)))
+
```

Comment:
I don't think you need that here. Annotation already contains a bunch of overloaded apply methods to create annotations; we should not add more overloadings in other places because then there are two places to look for the right variant instead of one. You can get waht you want already by writing

    addAnnotation(Annotation(cls, Nil))

If one objects to the `Nil`, one can add another variant of `apply` to `Annotation`.